### ISSUE TITLE
Webpack doesn't include AMD module dependencies if it uses an arrow function

### ISSUE THREAD

**Issue by bschlenk** [webpack/webpack#5916]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

If an AMD module is defined like this:

```js
define(require => {
    const dep = require('dep');
    ...
    const myModule = { ... };
    return myModule;
});
```

Then it will not also include `dep` in the bundle.

**If the current behavior is a bug, please provide the steps to reproduce.**

Create an entry file that is an AMD module, which uses an arrow function like the one shown above. Compile it using webpack. None of the dependencies of that file will be included in the bundle.

https://gist.github.com/bschlenk/35c66ca0183330e4790275f589d1353a

**What is the expected behavior?**

I would expect the dependencies of the AMD module to be included in the bundle. They do get included if the module is written using a regular function:

```js
define(function(require) {
    ...
});
```

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

webpack 3.7.1

**Comment by TheLarkInn** [comment]
This is probably parser bug isn't it

**Comment by sokra** [comment]
You can send a PR. It's somewhere in AMDParserPlugin

**Comment by bschlenk** [comment]
I think I found the issue: https://github.com/bschlenk/webpack/blob/master/lib/dependencies/AMDDefineDependencyParserPlugin.js#L41
This only checks FunctionExpressions, not ArrowFunctionExpressions. How would I go about testing a fix for this? I'm new to contributing to webpack. Ran the coverage tests to see which tests touch this module, but it doesn't look like this module is tested at all.

**Comment by sokra** [comment]
Here is an existing test for AMD: https://github.com/webpack/webpack/tree/master/test/cases/parsing/extract-amd

**Comment by bschlenk** [comment]
The code I have now is failing because it generates this:

```js
it("should allow AMD define with an arrow function", function(done) {
	!(__WEBPACK_AMD_DEFINE_RESULT__ = (require) => {
		__webpack_require__(/*! ./amdmodule */ 13).should.be.eql("a");
		done();
	}.call(exports, __webpack_require__, exports, module),
				__WEBPACK_AMD_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_AMD_DEFINE_RESULT__));
});
```

This is not valid javascript: () => {}.call()

What would make more sense here, wrapping the arrow function in parenthesis, which would work for both normal and arrow functions:

```js
(() => {}).call()

(function() => {}).call()
```

or converting any arrow functions to normal functions?

**Comment by bschlenk** [comment]
This was released in 3.9.0
